### ISSUE TITLE
Feature request: filter by antennaType

### ISSUE THREAD

**Issue by Kerbas-ad-astra** [Crzyrndm/FilterExtension#67]
The Science/Data Transmitter subcategory is overflowing with command pods and probe cores (because of their INTERNAL antennas), which are not really what I'm after when looking for "proper" antennas.  It would also be nice to have direct and relay antennas live in distinct categories.

**Comment by Crzyrndm** [comment]
latest commit should do the job
```
CHECK
{
	type = field
	value = ModuleDataTransmitter,antennaType,DIRECT
}
```
format is: module name, field name, field value

Not a huge amount of testing just yet and might be a little slow for the moment

**Comment by Kerbas-ad-astra** [comment]
That check will also be useful for excluding probes from the science category (I have Starwaster's Probe Science patch installed, but again, that's not what I'm after when I'm looking for experiments).  Thanks!

**Comment by Crzyrndm** [comment]
Yup, the idea is that this is about as generic as I can make it.

Current commit should fix inherited types not working (ModuleEngines matching all engines now) and speed it up a lot. Remaining shortcoming would be that it won't work well for numeric values (ie < this value)

**Comment by Kerbas-ad-astra** [comment]
I definitely notice the extra thinking time (is that mostly driven by the logging?) when selecting the transmitter and experiment subcategories (even on my test install with stock parts), and of course selecting the parent category gets the whammy from all of them.  However, not having to crawl through pods and probes when I'm looking for science parts is worth the brief delay!

**Comment by Crzyrndm** [comment]
If you still have yeterdays version, it's partially the logging and partially just how it was done. New stuff should be much less noticable, but still something to be careful with...

**Comment by Kerbas-ad-astra** [comment]
I used the DLL from version 6a37660.

Also, when I tested in my main (heavily-modded install), there was a nullref from the nodecheck when it attempted to check the experimentID of an Orbital Material Science experiment storage part.  The ExperimentStorage PartModule is a child of ModuleScienceExperiment, but it has a null experiment if it's not storing an experiment.  I know other addons have choked on this module and other experiment-storage modules from e.g. Wild Blue Industries and L-Tech and need extra null checks before e.g. trying to call for an experiment from an experimentID which may be null.

```
[Filter Extensions 2.7.1.0]: node check: NE.ESC.US.Wedge, ModuleScienceExperiment,experimentID,probeReport
[Filter Extensions 2.7.1.0]: valid params, type found ModuleScienceExperiment
[...]
[Filter Extensions 2.7.1.0]: Module type: NE_Science.ExperimentStorage, Is assignable: True
[Filter Extensions 2.7.1.0]: field: experimentID
NullReferenceException: Object reference not set to an instance of an object
  at FilterExtensions.Utility.PartType.NodeCheck (.AvailablePart part, System.String[] parameters) [0x00000] in <filename unknown>:0 
  at FilterExtensions.ConfigNodes.Check.checkResult (.AvailablePart part, Int32 depth) [0x00000] in <filename unknown>:0 
  at FilterExtensions.ConfigNodes.Filter.filterResult (.AvailablePart part, Int32 depth) [0x00000] in <filename unknown>:0 
  at FilterExtensions.ConfigNodes.customSubCategory.checkFilters (.AvailablePart ap, Int32 depth) [0x00000] in <filename unknown>:0 
  at FilterExtensions.ConfigNodes.customSubCategory.checkTemplate (.AvailablePart ap, Int32 depth) [0x00000] in <filename unknown>:0 
  at FilterExtensions.ConfigNodes.customSubCategory.checkPartFilters (.AvailablePart part, Int32 depth) [0x00000] in <filename unknown>:0 
  at FilterExtensions.ConfigNodes.customSubCategory.<initialise>b__25_0 (.AvailablePart p) [0x00000] in <filename unknown>:0 
  at EditorPartListFilter`1[AvailablePart].FilterList (System.Collections.Generic.List`1 parts, System.Func`2 filter) [0x00000] in <filename unknown>:0 
  at EditorPartListFilterList`1[AvailablePart].FilterList (System.Collections.Generic.List`1 list) [0x00000] in <filename unknown>:0 
  at KSP.UI.Screens.EditorPartList.RefreshPartList () [0x00000] in <filename unknown>:0 
  at KSP.UI.Screens.EditorPartList.Refresh () [0x00000] in <filename unknown>:0 
  at KSP.UI.Screens.EditorPartList.Refresh (State state) [0x00000] in <filename unknown>:0 
  at KSP.UI.Screens.PartCategorizer+<UpdateDaemon>c__Iterator127.MoveNext () [0x00000] in <filename unknown>:0 
  at UnityEngine.SetupCoroutine.InvokeMoveNext (IEnumerator enumerator, IntPtr returnValueAddress) [0x00000] in <filename unknown>:0 
```

**Comment by Crzyrndm** [comment]
Null check added

**Comment by Kerbas-ad-astra** [comment]
Looks good! I only notice the hangup on the first-time selection, and no exceptions are thrown.  I'll make a PR with my category changes soon.

**Comment by Crzyrndm** [comment]
Categorisation merged from #68. Any small fixes required to this feature?

**Comment by Kerbas-ad-astra** [comment]
Not as I recall -- I've been using those categories privately and found them quite convenient.
